src/corehost/common/pal.unix.cpp

```diff
@@ -57,6 +57,37 @@ bool pal::touch_file(const pal::string_t& path)
     return true;
 }
 
+void* pal::map_file_readonly(const pal::string_t& path, size_t& length)
+{
+	int fd = open(path.c_str(), O_RDONLY, (S_IRUSR | S_IRGRP | S_IROTH));
+	if (fd == -1)
+	{
+		trace::warning(_X("Failed to map file. open(%s) failed with error %d"), path.c_str(), errno);
+		return nullptr;
+	}
+
+	struct stat buf;
+	if (fstat(fd, &buf) != 0)
+	{
+		trace::warning(_X("Failed to map file. fstat(%s) failed with error %d"), path.c_str(), errno);
+        close(fd);
+        return nullptr;
+	}
+
+	length = buf.st_size;
+	void* address = mmap(nullptr, , PROT_READ, MAP_SHARED, fd, 0);
+
+	if(address == nullptr)
+	{
+		trace::warning(_X("Failed to map file. mmap(%s) failed with error %d"), path.c_str(), errno);
+        close(fd);
+        return nullptr;
+	}
+
+	close(fd);
```

Comment:
Hmm - here we close the file, on Windows we don't... is there a reason?

Reply:
The [manual](http://man7.org/linux/man-pages/man2/mmap.2.html) for `mmap()` says: "After the mmap() call has returned, the file descriptor, fd, can be closed immediately without invalidating the mapping." So, we release the file-descriptor after the mapping.

The [manual](https://docs.microsoft.com/en-us/windows/desktop/api/memoryapi/nf-memoryapi-createfilemappingw) for `CreateFileMapping` seems to say the same thing (now that I've read it again) but in a less straightforward way. I'll update the Windows code to close the file handle.